docs/docs/troubleshooting.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Troubleshooting
+description: >-
+  Learn how to troubleshoot common configuration issues or work around any outstanding bugs.
+sidebarDepth: 0
+lang: en-US
+meta:
+  - name: keywords
+    content: pomerium troubleshooting
+---
+
+# Troubleshooting
+
+This article provides troubleshooting information for various tools and features from Pomerium.
+
+[[toc]]
+
+## Pomerium Core
+
+### JWT Authentication with Let's Encrypt
+
+Wildcard certificates signed by LetsEncrypt may still be cross-signed by the [expired DST R3 root](https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/). While many browsers still trust these certificates (as long as they are also signed by a valid root), some upstream applications reject them, including Grafana as an example:
+
+```log
+logger=context error=Get "https://authenticate.localhost.pomerium.io/.well-known/pomerium/jwks.json": x509: certificate signed by unknown authority
+```
+
+For upstream applications that can use a local signing key file, you can circumvent this issue using `curl` or `wget` to download the signing key locally (relative to the upstream service):
+
+::::: tabs
+:::: tab curl
+
+```bash
+curl https://authenticate.localhost.pomerium.io/.well-known/pomerium/jwks.json > /etc/grafana/jwks.json
+```
+::::
+:::: tab wget
+
+```bash
+wget -O /etc/upstream-service/jwks.json https://authenticate.localhost.pomerium.io/.well-known/pomerium/jwks.json
+```
+::::
+:::::
+
+Edit the upstream service configuration to use the local key to to verify tokens.
+
+### Kubernetes Ingress Controller
+
+#### View Event History
+
+Pomerium Ingress Controller will add **events** to the Ingress objects as it processes them.
+
+```
+kubectl describe ingress/my-ingress
+```
+
+```log
+Events:
+  Type    Reason   Age   From              Message
+  ----    ------   ----  ----              -------
+  Normal  Updated  18s   pomerium-ingress  updated pomerium configuration
+```
+
+If an error occurs, it may be reflected in the events:
+
+```log
+Events:
+  Type     Reason       Age                 From              Message
+  ----     ------       ----                ----              -------
+  Normal   Updated      5m53s               pomerium-ingress  updated pomerium configuration
+  Warning  UpdateError  3s                  pomerium-ingress  upsert routes: parsing ingress: annotations: applying policy annotations: parsing policy: invalid rules in policy: unsupported conditional "maybe", only and, or, not, nor and action are allowed
+```
+
+#### HSTS
```

Comment:
- We can pull up HSTS to a higher level vs just being k8s based
- Reference the full header name here too `Strict-Transport-Security` too for search ability
- We should probably update the reference docs where this is set (https://www.pomerium.io/reference/#set-response-headers) to mention and link to (I think Mozilla does a really nice job here) each of the default security headers we set by default. (e.g. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security)

Reply:
Troubleshooting page adjuster in https://github.com/pomerium/pomerium/pull/2797/commits/58101e1bda4b42e22b6d944e8d0da8f9b577a6bb
Reference page adjusted in https://github.com/pomerium/pomerium/pull/2797/commits/6cc4df64fb5272d0804e3aafe5ea0049e5c1c590